runtime-sdk/modules/evm/src/lib.rs

```diff
@@ -0,0 +1,361 @@
+        )
+    }
+
+    fn tx_call<C: TxContext>(ctx: &mut C, body: types::CallTx) -> Result<Vec<u8>, Error> {
+        Self::call(
+            ctx,
+            H160::from_slice(&body.address),
+            U256::from_big_endian(&body.value),
+            body.data,
+            U256::from_big_endian(&body.gas_price),
+            body.gas_limit,
+        )
+    }
+
+    fn q_peek_storage<C: Context>(
+        ctx: &mut C,
+        body: types::PeekStorageQuery,
+    ) -> Result<Vec<u8>, Error> {
+        Self::peek_storage(
+            ctx,
+            H160::from_slice(&body.address),
+            H256::from_slice(&body.index),
+        )
+    }
+
+    fn q_peek_code<C: Context>(ctx: &mut C, body: types::PeekCodeQuery) -> Result<Vec<u8>, Error> {
+        Self::peek_code(ctx, H160::from_slice(&body.address))
+    }
+}
+
+impl module::MethodHandler for Module {
+    fn dispatch_call<C: TxContext>(
+        ctx: &mut C,
+        method: &str,
+        body: cbor::Value,
+    ) -> module::DispatchResult<cbor::Value, CallResult> {
+        match method {
+            "evm.Create" => {
+                let result = || -> Result<cbor::Value, Error> {
+                    let args = cbor::from_value(body).map_err(|_| Error::InvalidArgument)?;
+                    Ok(cbor::to_value(Self::tx_create(ctx, args)?))
+                }();
+                match result {
+                    Ok(value) => module::DispatchResult::Handled(CallResult::Ok(value)),
+                    Err(err) => module::DispatchResult::Handled(err.to_call_result()),
+                }
+            }
+            "evm.Call" => {
+                let result = || -> Result<cbor::Value, Error> {
+                    let args = cbor::from_value(body).map_err(|_| Error::InvalidArgument)?;
+                    Ok(cbor::to_value(Self::tx_call(ctx, args)?))
+                }();
+                match result {
+                    Ok(value) => module::DispatchResult::Handled(CallResult::Ok(value)),
+                    Err(err) => module::DispatchResult::Handled(err.to_call_result()),
+                }
+            }
+            _ => module::DispatchResult::Unhandled(body),
+        }
+    }
+
+    fn dispatch_query<C: Context>(
+        ctx: &mut C,
+        method: &str,
+        args: cbor::Value,
+    ) -> module::DispatchResult<cbor::Value, Result<cbor::Value, error::RuntimeError>> {
+        match method {
+            "evm.PeekStorage" => module::DispatchResult::Handled((|| {
+                let args = cbor::from_value(args).map_err(|_| Error::InvalidArgument)?;
+                Ok(cbor::to_value(Self::q_peek_storage(ctx, args)?))
+            })()),
+            "evm.PeekCode" => module::DispatchResult::Handled((|| {
+                let args = cbor::from_value(args).map_err(|_| Error::InvalidArgument)?;
+                Ok(cbor::to_value(Self::q_peek_code(ctx, args)?))
+            })()),
+            _ => module::DispatchResult::Unhandled(args),
+        }
+    }
+}
```

Comment:
sourcecode boil, sourcecode bubble, something plate-y this way comes. I wonder if #139 will ever land?

Reply:
Would be great if it did ;-)